src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java

```diff
@@ -37,465 +24,389 @@
         if (CollectionUtils.isNotEmpty(submitEvents)) {
-            executeReport(listingDetails, submitEvents);
+            executeReport(submitEvents, reportData);
         }
+        return reportData;
+    }
 
-        listingDetails.getCaseData().clearReportFields();
-        return listingDetails.getCaseData();
+    private HearingsByHearingTypeReportData initReport(String caseTypeId) {
+        var reportSummaryHdr = new HearingsByHearingTypeReportSummaryHdr();
+        var fields = new ReportFields();
+        initReportFields(fields);
+        reportSummaryHdr.setOffice(UtilHelper.getListingCaseTypeId(caseTypeId));
+        reportSummaryHdr.setFields(fields);
+        return new HearingsByHearingTypeReportData(reportSummaryHdr);
     }
 
-    private void executeReport(ListingDetails listingDetails, List<SubmitEvent> submitEvents) {
-        log.info(String.format("Hearings by hearing type report case type id %s search results: %d",
-                listingDetails.getCaseTypeId(), submitEvents.size()));
-        if (UtilHelper.getListingCaseTypeId(listingDetails.getCaseTypeId()).toLowerCase().contains("scotland")) {
-            costsHearingType = "Expenses/Wasted Costs Hearing";
+    private void initReportFields(ReportFields reportFields) {
+        reportFields.setHearingCount("0");
+        reportFields.setCmCount("0");
+        reportFields.setCostsCount("0");
+        reportFields.setTotal("0");
+        reportFields.setHearingPrelimCount("0");
+        reportFields.setReconsiderCount("0");
+        reportFields.setRemedyCount("0");
+    }
+
+    private boolean isValidHearing(DateListedType dateListedType) {
+        LocalDate listedDate;
+        var startDate = LocalDate.parse(ReportHelper.getFormattedLocalDate(dateFrom));
+        var endDate = LocalDate.parse(ReportHelper.getFormattedLocalDate(dateTo));
+        if (isNullOrEmpty(dateListedType.getListedDate())) {
+            return false;
+        } else {
+            listedDate = LocalDate.parse(ReportHelper.getFormattedLocalDate(dateListedType.getListedDate()));
         }
-        populateLocalReportSummaryHdr(listingDetails.getCaseData(), submitEvents);
-        populateLocalReportSummary(listingDetails.getCaseData(), submitEvents);
-        populateLocalReportSummaryHdr2(listingDetails.getCaseData(), submitEvents);
-        populateLocalReportSummary2(listingDetails.getCaseData(), submitEvents);
-        populateLocalReportSummaryDetail(listingDetails.getCaseData(), submitEvents);
-
-        if (CollectionUtils.isNotEmpty(listingDetails.getCaseData().getLocalReportsSummary())) {
-            listingDetails.getCaseData().getLocalReportsSummary().get(0).getValue().setReportOffice(
-                    UtilHelper.getListingCaseTypeId(listingDetails.getCaseTypeId()));
+        if (HEARING_STATUS_HEARD.equals(dateListedType.getHearingStatus())) {
+            return (listedDate.isEqual(startDate) || listedDate.isAfter(startDate))
```

Comment:
Can you simplify this? !listedDate.before(startDate) && !listDate.after(endDate) ?

Reply:
refactored